image/descriptor.go

```diff
@@ -66,22 +79,41 @@ func findDescriptor(w walker, name string) (*v1.Descriptor, error) {
 			return err
 		}
 
-		for i := 0; i < len(index.Manifests); i++ {
-			if index.Manifests[i].Annotations["org.opencontainers.ref.name"] == name {
-				d = index.Manifests[i]
-				return errEOW
+		switch argsParts[0] {
+		case "org.opencontainers.ref.name":
+			for i := 0; i < len(index.Manifests); i++ {
+				if index.Manifests[i].Annotations["org.opencontainers.ref.name"] == argsParts[1] {
+					refs = append(refs, index.Manifests[i])
+				}
+			}
+		case "platform.os":
+			for i := 0; i < len(index.Manifests); i++ {
+				if index.Manifests[i].Platform != nil && index.Manifests[i].Platform.OS == argsParts[1] {
+					refs = append(refs, index.Manifests[i])
+				}
 			}
+		case "digest":
```

Comment:
I don't think so. when you get a given digest, we also need  `validateDescriptor` to verify that it conforms to the specification.